Source/BugsnagSink.m

```diff
@@ -59,9 +60,38 @@ - (void)filterReports:(NSArray *)reports
          onCompletion:(BSG_KSCrashReportFilterCompletion)onCompletion {
     NSMutableArray *bugsnagReports = [NSMutableArray new];
     BugsnagConfiguration *configuration = [Bugsnag configuration];
+    
     for (NSDictionary *report in reports) {
-        BugsnagCrashReport *bugsnagReport =
-                [[BugsnagCrashReport alloc] initWithKSReport:report];
+        BugsnagCrashReport *bugsnagReport = [[BugsnagCrashReport alloc] initWithKSReport:report];
+        BOOL incompleteReport = (![@"standard" isEqualToString:[report valueForKeyPath:@"report.type"]] ||
+                                 [[report objectForKey:@"incomplete"] boolValue]);
+        
+        if (incompleteReport) { // append app/device data as this is unlikely to change between sessions
+            NSDictionary *sysInfo = [BSG_KSSystemInfo systemInfo];
+            
+            // reset any existing data as it will be corrupted/nil
+            bugsnagReport.appState = @{};
+            bugsnagReport.deviceState = @{};
+            
+            bugsnagReport.app = @{
+                                  @"bundleVersion": sysInfo[@BSG_KSSystemField_BundleVersion],
+                                  @"id": sysInfo[@BSG_KSSystemField_BundleID],
+                                  @"releaseStage": configuration.releaseStage,
```

Comment:
There is a risk here for any of these values to be nil, particularly release stage if overridden by the user, so inserting values only if not nil would be prudent here as this will otherwise crash if any of these values is not set.